src/test/java/it/mulders/mcs/search/vulnerability/ComponentReportClientIT.java

```diff
@@ -0,0 +1,128 @@
+package it.mulders.mcs.search.vulnerability;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.ConnectException;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.List;
+
+import com.github.tomakehurst.wiremock.junit5.WireMockRuntimeInfo;
+import com.github.tomakehurst.wiremock.junit5.WireMockTest;
+import it.mulders.mcs.common.Result;
+import org.assertj.core.api.WithAssertions;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.DisplayNameGeneration;
+import org.junit.jupiter.api.DisplayNameGenerator;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.badRequest;
+import static com.github.tomakehurst.wiremock.client.WireMock.ok;
+import static com.github.tomakehurst.wiremock.client.WireMock.post;
+import static com.github.tomakehurst.wiremock.client.WireMock.stubFor;
+import static com.github.tomakehurst.wiremock.client.WireMock.urlPathMatching;
+
+@DisplayNameGeneration(DisplayNameGenerator.ReplaceUnderscores.class)
+@WireMockTest
+class ComponentReportClientIT implements WithAssertions {
+  String getResourceAsString(final String resourceName) {
+    try (final InputStream input = getClass().getResourceAsStream(resourceName)) {
+      byte[] bytes = input != null ? input.readAllBytes() : new byte[]{};
+        return new String(bytes, StandardCharsets.UTF_8);
+
+    } catch (final IOException ioe) {
+      return fail("Can't load resource %s", resourceName, ioe);
+    }
+  }
+
+  @Nested
+  @DisplayName("ComponentReport with vulnerabilities")
+  class ComponentReportWithVulnerabilitiesTest {
+    @BeforeEach
+    void setup() {
+      System.setProperty("ossindex.username", "user");
```

Comment:
What do you think of using `@SetSystemProperty` from [JUnit Pioneer Extensions](https://junit-pioneer.org/docs/system-properties/) for this purpose?

Reply:
> What do you think of using `@SetSystemProperty` from [JUnit Pioneer Extensions](https://junit-pioneer.org/docs/system-properties/) for this purpose?

01417039